### DISCUSSION TITLE
cmake minimum version requirement for rpm?

### DISCUSSION THREAD

**Discussion by pmatilai** [rpm-software-management/rpm#2248] [General]
As we're moving to cmake in the next major release, here's a bit of a community poll: what would be a reasonable minimum version to require?

Rpm is conservative in its requirements so it's not like we'd require a cmake from last week or even last year, but AIUI  cmake is very different from autotools in that things are not expected to break left and right when you update. 

I initially tossed in version 3.10 as minimum as there were some important looking changes around that time, but it's starting to seem long in the tooth and overly conservative. Buildability on latest RHEL is one of our unwritten requirements, but RHEL 9 and even 8 have cmake 3.20 already. 

Thoughts from those more familiar with the cmake ecosystem, from other distros perspective etc?

**Comment by DemiMarie** [comment]
It should be possible to build on Debian stable.

**Comment by oliviacrain** [comment]
Speaking with my [CBL-Mariner](https://github.com/microsoft/CBL-Mariner) distro maintainer hat on- no objections to having a floor of cmake 3.20. The lowest version of cmake we support in Mariner at this time is 3.21.4.

**Comment by pmatilai** [comment]
Debian isn't exactly a prime target for rpm :smile: but for the record, it seems to be at 3.18.4 from 2020, which is probably newer than rpm can reasonably require. So that should be easily covered. 

In the meanwhile I realized rpm is already relying on at least >= 3.12 despite what our CMakeLists.txt (for deitys sake who came up with that filename...) claims. 3.12 is from 2018 and that may be more like it.

**Reply by Conan-Kudo** [reply]
I used 3.12 for popt because that's the minimum version with helper functions for easily creating the cmake config/version files for `find_package()`. I'm fine with this as the minimum for rpm too.

(I was going to bump it anyway when I synced my improvements from popt into rpm 😅 )

**Comment by pmatilai** [comment]
Ended up bumping it to 3.18 after running into enough more or less minor but nice stuff. That's as far as it can go though.
